arangod/VocBase/LogicalCollection.cpp

```diff
@@ -357,20 +357,24 @@ replication::Version LogicalCollection::replicationVersion() const noexcept {
 }
 
 std::string const& LogicalCollection::distributeShardsLike() const noexcept {
-  TRI_ASSERT(_sharding != nullptr);
-  return _sharding->distributeShardsLike();
-}
-
-void LogicalCollection::distributeShardsLike(std::string const& cid,
-                                             ShardingInfo const* other) {
-  TRI_ASSERT(_sharding != nullptr);
-  _sharding->distributeShardsLike(cid, other);
-}
-
-std::vector<std::string> const& LogicalCollection::avoidServers()
-    const noexcept {
-  TRI_ASSERT(_sharding != nullptr);
-  return _sharding->avoidServers();
+  if (ServerState::instance()->isCoordinator() &&
+      replicationVersion() == replication::Version::TWO) {
+    auto& cf = vocbase().server().getFeature<ClusterFeature>();
+    auto& ci = cf.clusterInfo();
+
+    auto myGroup = ci.getCollectionGroupById(groupID());
+    TRI_ASSERT(myGroup != nullptr)
+        << "Collection part of a Group that does not exist";
```

Comment:
Can this be a race? E.g., can the LogicalCollection object still be around for a while after the collection group has already been deleted, and is thus missing from ClusterInfo?

Reply:
Good point I think it actually can.
Will add graceful handling.